Source/ACE.Database/Adapter/BiotaConverter.cs

```diff
@@ -527,6 +527,33 @@ public static ACE.Database.Models.Shard.Biota ConvertFromEntityBiota(ACE.Entity.
                 {
                     var entity = new BiotaPropertiesPosition { ObjectId = biota.Id, PositionType = (ushort)kvp.Key, ObjCellId = kvp.Value.ObjCellId, OriginX = kvp.Value.PositionX, OriginY = kvp.Value.PositionY, OriginZ = kvp.Value.PositionZ, AnglesW = kvp.Value.RotationW, AnglesX = kvp.Value.RotationX, AnglesY = kvp.Value.RotationY, AnglesZ = kvp.Value.RotationZ };
 
+                    // Entity Framework is unable to store NaN floats in the database and results in an error of:
+                    // ERROR 1054: Unknown column 'NaN' in 'field list'
+
+                    var isNaNCount = 0;
+                    if (float.IsNaN(entity.AnglesX))
+                    {
+                        entity.AnglesX = 0;
```

Comment:
if only some of these values are adjusted, but not others, you could end up with a non-normalized quaternion, which would be very bad

a better strategy might be if any fields are NaN, then to set the quaternion to identity. this would match the same strategy used by the acclient physics engine